File: Interview Prep/Sales Checker/SalesChecker.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func solution(M, R, D int) string {
	initialSales := min(M, D)
	remainingDays := D - initialSales
	additionalSales := remainingDays
	maxSales := initialSales + additionalSales
	if maxSales >= R {
		return "YES"
	}
	return "NO"
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter the number of test cases: ")
	tcStr, _ := reader.ReadString('\n')
	tc, _ := strconv.Atoi(strings.TrimSpace(tcStr))
	for t := 0; t < tc; t++ {
		line, _ := reader.ReadString('\n')
		parts := strings.Fields(line)
		M, _ := strconv.Atoi(parts[0])
		R, _ := strconv.Atoi(parts[1])
		D, _ := strconv.Atoi(parts[2])
		result := solution(M, R, D)
		fmt.Println(result)
	}
}

File: Interview Prep/Sales Checker/SalesChecker.cpp
#include <iostream>
#include <string>
using namespace std;

string solution(int M, int R, int D) {
  int initialSales = min(M, D);
  int remainingDays = D - initialSales;
  int additionalSales = remainingDays;
  int maxSales = initialSales + additionalSales;
  return maxSales >= R ? "YES" : "NO";
}

int main() {
  int tc;
  cin >> tc;
  while (tc-- > 0) {
    int M, R, D;
    cin >> M >> R >> D;
    string result = solution(M, R, D);
    cout << result << endl;
  }
  return 0;
}